### PR TITLE
Do not suppress diff if it is explicitly changed to zero

### PR DESCRIPTION
**Pull request by andrewnester**
## Changes
Do not suppress diff if it is explicitly changed to zero

## Tests
Added regression tests

- [ ] `make test` run locally
- [ ] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

**Review by pietern — COMMENTED** [review]
LGTM, but will have to defer to @alexott and @tanmay-db for a stamp.

Diff suppression is a beast and there could be side effects.

File: common/reflect_resource.go
Comment (tanmay-db) [inline common/reflect_resource.go:371]: I think we should also have this in `SetSuppressDiffWithDefault` as well: https://github.com/databricks/terraform-provider-databricks/blob/main/common/customizable_schema.go#L159. 

````
	// Suppress diff if the new value (ie value from HCL config) is not set and
	// the old value (ie value from state / platform) is equal to the default value.
	s.Schema.DiffSuppressFunc = func(k, old, new string, d *schema.ResourceData) bool {
		if new == zero && old == sv {
			return true
		}
		return false
	}
````

Otherwise LGTM, stamping to unblock merge.

**Review by tanmay-db — APPROVED** [review]

**Review by alexott — APPROVED** [review]

**Pull request merged**
